metrics.go

```diff
@@ -76,12 +89,28 @@ func (m *Metrics) MeasureSinceWithLabels(key []string, start time.Time, labels [
 	if m.ServiceName != "" {
 		key = insert(0, m.ServiceName, key)
 	}
+	if !m.allowMetric(key) {
+		return
+	}
 	now := time.Now()
 	elapsed := now.Sub(start)
 	msec := float32(elapsed.Nanoseconds()) / float32(m.TimerGranularity)
 	m.sink.AddSampleWithLabels(key, msec, labels)
 }
 
+// Returns whether the metric should be allowed based on configured prefix filters
+func (m *Metrics) allowMetric(key []string) bool {
+	if m.filter == nil || m.filter.Len() == 0 {
+		return true
```

Comment:
This should probably return `m.Config.FilterDefault` so you could block all with that `false` and an empty filter.